ui/jquery.ui.mouse.js

```diff
@@ -21,12 +21,137 @@
 ///////////////////////////////////////////////////////////////////////
 
 //JASPERSOFT #1 #3
-define(function(require) {
+define(function (require) {
 
 	var $ = require("./jquery.ui.widget");
+
 	var isIE = function () {
 		return navigator.appName === "Microsoft Internet Explorer" || navigator.userAgent.indexOf("Trident/") >= 0;
 	};
+
+	var detectedIEVersion = false;
+
+	var getScrollbarWidth = function () {
+		var outer = document.createElement("div");
+		outer.style.visibility = "hidden";
+		outer.style.width = "100px";
+		outer.style.msOverflowStyle = "scrollbar"; // needed for WinJS apps
+
+		document.body.appendChild(outer);
+
+		var widthNoScroll = outer.offsetWidth;
+		// force scrollbars
+		outer.style.overflow = "scroll";
+
+		// add innerdiv
+		var inner = document.createElement("div");
+		inner.style.width = "100%";
+		outer.appendChild(inner);
+
+		var widthWithScroll = inner.offsetWidth;
+
+		// remove divs
+		outer.parentNode.removeChild(outer);
+
+		return widthNoScroll - widthWithScroll;
+	};
+
+	var getIEVersion = function () {
+		var version = 0;
+		if (detectedIEVersion !== false) {
+			return detectedIEVersion;
+		}
+
+		if (this.isIE()) {
+			if (navigator.appName === "Netscape") {
+				var ua = navigator.userAgent;
+				var re = new RegExp("Trident/.*rv:([0-9]{1,}[\\.0-9]{0,})");
+				if (re.exec(ua) != null) {
+					version = parseFloat(RegExp.$1);
+				}
+			} else {
+				var msVersion = navigator.appVersion.split("MSIE")[1];
+				version = parseFloat(msVersion);
+			}
+		}
+
+		detectedIEVersion = version;
+		return version;
+	};
+
+	var isIE11 = function () {
+		return this.getIEVersion() === 11;
+	};
+	var inRect = function (rect, x, y) {
+		return (y >= rect.top && y <= rect.bottom) && (x >= rect.left && x <= rect.right);
+	};
+
+	var hasScroll = function ($element, axis) {
+		var overflow = $element.css("overflow"),
+			overflowAxis;
+
+		if (typeof axis === "undefined" || axis === "y") {
+			overflowAxis = $element.css("overflow-y");
+		} else {
+			overflowAxis = $element.css("overflow-x");
+		}
+
+		var bShouldScroll = $element.get(0).scrollHeight > $element.innerHeight();
+
+		var bAllowedScroll = (overflow === "auto" || overflow === "visible") ||
+			(overflowAxis === "auto" || overflowAxis === "visible");
+
+		var bOverrideScroll = overflow === "scroll" || overflowAxis === "scroll";
+
+		return (bShouldScroll && bAllowedScroll) || bOverrideScroll;
+	};
+
+	var inScrollRange = function (event) {
+
+		var
+			scrollSize = getScrollbarWidth(),
```

Comment:
You are calculating scrollbar with each time on mouse move.
Scrollbar calc is heavy operation since it involves few DOM operations.
I suggest to calculate scrollbar width just once when jquery library is loaded

Reply:
right :+1: (
thank you for catching this one, fixing it...

Reply:
Fixed